src/datasources/tag/TagDataSource.ts

```diff
@@ -25,94 +35,222 @@ export class TagDataSource extends DataSourceBase<TagQuery> {
+      // Merge the results from the current chunk with the aggregated results
+      for (const [path, data] of Object.entries(chunkResult.results)) {
+        if (!aggregatedResults.results[path]) {
+          aggregatedResults.results[path] = data;
+        } else {
+          aggregatedResults.results[path].values = aggregatedResults.results[path].values.concat(data.values);
+        }
+      }
+    }
+
+    return aggregatedResults;
   }
 
-  private async getTagHistoryValues(path: string, workspace: string, range: TimeRange, intervals?: number) {
-    const response = await this.post<TagHistoryResponse>(this.tagHistoryUrl + '/query-decimated-history', {
-      paths: [path],
-      workspace,
+  getTagHistoryValues = async (paths: string[], workspace: string, range: TimeRange, intervals?: number): Promise<TagHistoryResponse> => {
+    return this.post<TagHistoryResponse>(`${this.tagHistoryUrl}/query-decimated-history`, {
+      paths: paths,
+      workspace: workspace,
       startTime: range.from.toISOString(),
       endTime: range.to.toISOString(),
       decimation: intervals ? Math.min(intervals, 1000) : 500,
     });
+  };
 
-    const { type, values } = response.results[path];
-    return {
-      datetimes: values.map(v => dateTime(v.timestamp).valueOf()),
-      values: values.map(v => this.convertTagValue(type, v.value)),
-    };
-  }
-
-  private convertTagValue(type: string, value?: string) {
-    return value && ['DOUBLE', 'INT', 'U_INT64'].includes(type) ? Number(value) : value;
+  private convertTagValue(type: TagDataType, value?: string) {
+    return value && [TagDataType.DOUBLE, TagDataType.INT, TagDataType.U_INT64].includes(type) ? Number(value) : value;
   }
 
-  private getPropertiesAsFields(properties: Record<string, string> | null): FieldDTO[] {
-    if (!properties) {
-      return [];
-    }
+  private getAllProperties(data: TagWithValue[]) {
+    let props: Set<string> = new Set();
+    data.forEach((tag) => {
+      if (tag.tag.properties) {
+        Object.keys(tag.tag.properties)
+          .filter(name => !name.startsWith('nitag'))
+          .forEach((name) => {
+            props.add(name)
+          })
+      }
+    })
 
-    return Object.keys(properties)
-      .filter(name => !name.startsWith('nitag'))
-      .map(name => ({ name, values: [properties[name]] }));
+    return props
   }
 
   shouldRunQuery(query: TagQuery): boolean {
     return Boolean(query.path);
   }
 
+  async metricFindQuery({ workspace, path }: TagVariableQuery): Promise<MetricFindValue[]> {
+    // Skip querying when no paths are provided
+    if (!path) {
+      return []
+    }
+    let paths: string[] = [path];
+    // wrap replaced variable in extra curly braces, protecting original {} tags from misinterpretation.
+    let parsed_workspace = this.templateSrv.replace(workspace,);
+    if (this.templateSrv.containsTemplate(path)) {
+      let replaced_path = this.templateSrv.replace(path, {}, (v: string) => `{{${v}}}`);
```

Comment:
Can you elaborate more on this and the comment above? Why do we need to add two sets of brackets?